src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs

```diff
@@ -764,6 +764,91 @@ private static ICalibratorTrainer GetCalibratorTrainerOrThrow(IExceptionContext
             return new OneVersusAllTrainer(env, est, labelColumnName, imputeMissingLabelsAsNegative, GetCalibratorTrainerOrThrow(env, calibrator), maximumCalibrationExampleCount, useProbabilities);
         }
 
+        /// <summary>
+        /// Create a <see cref="OneVersusAllTrainer{TModelOut}"/>, which predicts a multiclass target using one-versus-all strategy with
+        /// the binary classification estimator specified by <paramref name="binaryEstimator"/>. This method works with binary classifiers that
+        /// are either already calibrated, or non calibrated ones you don't want calibrated. If you need to have your classifier calibrated, use the
+        /// <see cref="OneVersusAllUnCalibratedToCalibrated{TModelIn, TCalibrator}"/> method instead.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// In one-versus-all strategy, a binary classification algorithm is used to train one classifier for each class,
+        /// which distinguishes that class from all other classes. Prediction is then performed by running these binary classifiers,
+        /// and choosing the prediction with the highest confidence score.
+        /// </para>
+        /// </remarks>
+        /// <param name="catalog">The multiclass classification catalog trainer object.</param>
+        /// <param name="binaryEstimator">An instance of a binary <see cref="ITrainerEstimator{TTransformer, TPredictor}"/> used as the base trainer.</param>
+        /// <param name="labelColumnName">The name of the label column.</param>
+        /// <param name="imputeMissingLabelsAsNegative">Whether to treat missing labels as having negative labels, instead of keeping them missing.</param>
+        /// <param name="useProbabilities">Use probabilities (vs. raw outputs) to identify top-score category.</param>
+        /// <typeparam name="TModel">The type of the model. This type parameter will usually be inferred automatically from <paramref name="binaryEstimator"/>.</typeparam>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        ///  [!code-csharp[OneVersusAll](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/MulticlassClassification/OneVersusAll.cs)]
+        /// ]]></format>
+        /// </example>
+        public static OneVersusAllTrainer<TModel> OneVersusAllTyped<TModel>(this MulticlassClassificationCatalog.MulticlassClassificationTrainers catalog,
```

Comment:
Have you tested the new OneVersusAllTyped to see that you can match the scores of the existing one?

Reply:
Yep! I added and modified the tests in the `OvaTest.cs` to show that the scores match exactly.

Reply:
The test which directly compares the score of the non-typed to the newly created typed version, is the perfect unit test. Thanks.